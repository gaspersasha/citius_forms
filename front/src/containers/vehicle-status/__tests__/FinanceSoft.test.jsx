import React from 'react';
import { FinanceSoft } from '..';

const props = {
  productAdvertId: 3332415,
  financeQuoteId: 3735215,
  softCreditSearch: {
    band: 'A',
    score: 550,
  },
  typoProps: {
    title: { align: 'left', type: 'h3', className: '' },
    text: { align: 'left', type: 'p', className: '' },
  },
};

describe('<FinanceSoft />', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<FinanceSoft {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should correctly display probability text #1', () => {
    const wrapper = shallow(<FinanceSoft {...props} />);

    expect(
      wrapper.find('[data-id="probability-text"]').prop('children')
    ).toEqual([
      'From your soft credit check we can see that you are',
      ' very likely ',
      'to be approved for finance',
    ]);
  });

  it('should correctly display probability text #2', () => {
    const wrapper = shallow(
      <FinanceSoft
        {...props}
        softCreditSearch={{
          ...props.softCreditSearch,
          band: 'C',
        }}
      />
    );

    expect(
      wrapper.find('[data-id="probability-text"]').prop('children')
    ).toEqual([
      'From your soft credit check we can see that you are',
      ' likely ',
      'to be approved for finance',
    ]);
  });
});
