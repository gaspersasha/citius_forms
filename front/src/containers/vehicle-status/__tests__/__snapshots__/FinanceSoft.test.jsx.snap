// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FinanceSoft /> should match snapshot 1`] = `
<Fragment>
  <Typography
    align="left"
    className=""
    color="dark"
    type="h3"
  >
    Your finance eligibility
  </Typography>
  <Typography
    align="left"
    className=""
    color="dark"
    data-id="probability-text"
    type="p"
  >
    From your soft credit check we can see that you are
     very likely 
    to be approved for finance
  </Typography>
  <Button
    className="btn"
    disabled={false}
    href="/finance?productAdvertId=3332415&financeQuoteId=3735215"
    onClick={null}
    selected={false}
    styleType="primary"
    target="_blank"
  >
    Continue your application
  </Button>
</Fragment>
`;
